### PR TITLE
 [SCB-591] configcenter need to do encode

### PR DESCRIPTION
**Pull request by weichao666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
huawei's huangyun set up an agent to modify the special characters '#'，which led to the governance inoperative

### CODE REVIEW

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java [hunk 1]
```diff
@@ -329,8 +329,14 @@ private void sendHeartbeat(WebSocket ws) {
 
     public void refreshConfig(String configcenter, boolean wait) {
       CountDownLatch latch = new CountDownLatch(1);
+      String encodeServiceName = "";
+      try {
+        encodeServiceName = URLEncoder.encode(serviceName, "UTF-8");
+      } catch (UnsupportedEncodingException e1) {
+        e1.printStackTrace();
```
Comment (wujimin) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: print?

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java [hunk 2]
```diff
@@ -329,8 +329,14 @@ private void sendHeartbeat(WebSocket ws) {
...
+      }
+      String path = URIConst.ITEMS + "?dimensionsInfo=" + StringUtils.deleteWhitespace(encodeServiceName);
```
Comment (wujimin) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: delete whitespace before or after encode?

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java (hunk 1 above)
Comment (weichao666) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java (hunk 2 above)
Comment (weichao666) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java [hunk 3]
```diff
@@ -329,8 +329,14 @@ private void sendHeartbeat(WebSocket ws) {
 
     public void refreshConfig(String configcenter, boolean wait) {
       CountDownLatch latch = new CountDownLatch(1);
+      String encodeServiceName = "";
+      try {
+        encodeServiceName = URLEncoder.encode(StringUtils.deleteWhitespace(serviceName), "UTF-8");
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.error("encode error: {}",e.getMessage());
```
[the remark this answers is not in the document]
Comment (liubao68) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: we can encoding serviceName in instance initialization if possible.

[the remark this answers is not in the document]
Comment (weichao666) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java]: ServiceName is stitched together in the CONFIG_CENTER_CONFIG.getServiceName() method，so you mean do encode in this method?

**Pull request merged**